tests/DateScalarTest.php

```diff
@@ -0,0 +1,130 @@
+<?php declare(strict_types=1);
+
+namespace MLL\GraphQLScalars\Tests;
+
+use DateTimeImmutable;
+use Generator;
+use GraphQL\Error\Error;
+use GraphQL\Error\InvariantViolation;
+use GraphQL\Language\AST\IntValueNode;
+use GraphQL\Language\AST\StringValueNode;
+use MLL\GraphQLScalars\DateScalar;
+use PHPUnit\Framework\TestCase;
+
+abstract class DateScalarTest extends TestCase
+{
+    /**
+     * @dataProvider invalidDateValues
+     *
+     * @param mixed $value An invalid value for a date
+     */
+    public function testThrowsIfSerializingInvalidDates($value): void
+    {
+        $this->expectException(InvariantViolation::class);
+
+        $this->scalarInstance()->serialize($value);
+    }
+
+    /**
+     * @dataProvider invalidDateValues
+     *
+     * @param mixed $value An invalid value for a date
+     */
+    public function testThrowsIfParseValueInvalidDate($value): void
+    {
+        $this->expectException(Error::class);
+
+        $this->scalarInstance()->parseValue($value);
+    }
+
+    /**
+     * Those values should fail passing as a date.
+     *
+     * @return Generator<array-key, array{mixed}>
+     */
+    public function invalidDateValues(): Generator
+    {
+        yield [1];
+        yield ['rofl'];
```

Comment:
```suggestion
        yield ['rolf'];
        yield ['rofl'];
```

I am fine with the addition, but we should keep `rolf` as the prototypical invalid value for MLL.